cmd/mattermost/commands/init.go

```diff
@@ -28,24 +28,31 @@ func initDBCommandContextCobra(command *cobra.Command, readOnlyConfigStore bool)
 }
 
 func InitDBCommandContextCobra(command *cobra.Command) (*app.App, error) {
-	return initDBCommandContextCobra(command, true)
+	return initDBCommandContextCobra(command, true, true)
 }
 
 func InitDBCommandContextCobraReadWrite(command *cobra.Command) (*app.App, error) {
-	return initDBCommandContextCobra(command, false)
+	return initDBCommandContextCobra(command, false, true)
 }
 
-func initDBCommandContext(configDSN string, readOnlyConfigStore bool) (*app.App, error) {
+func InitDBCommandContextCobraWithoutMetrics(command *cobra.Command) (*app.App, error) {
+	return initDBCommandContextCobra(command, true, false)
+}
+
+func initDBCommandContext(configDSN string, readOnlyConfigStore, startMetrics bool) (*app.App, error) {
 	if err := utils.TranslationsPreInit(); err != nil {
 		return nil, err
 	}
 	model.AppErrorInit(i18n.T)
 
-	s, err := app.NewServer(
-		// The option order is important as app.Config option reads app.StartMetrics option.
-		app.StartMetrics,
-		app.Config(configDSN, readOnlyConfigStore, nil),
-	)
+	options := []app.Option{}
+	if startMetrics {
+		options = append(options, app.StartMetrics)
+	}
+
+	// The option order is important as app.Config option reads app.StartMetrics option.
+	options = append(options, app.Config(configDSN, readOnlyConfigStore, nil))
```

Comment:
I think we might also need to pass `app.SkipPostInitialization()`. Can you check that? In that case, I think it might be simpler to not explode the number of functions further but rather pass the options from the caller side itself.